src/jdk.crypto.ec/share/classes/sun/security/ec/XDHPublicKeyImpl.java

```diff
@@ -107,6 +113,10 @@ public String getAlgorithm() {
         return "XDH";
     }
 
+    public Optional<byte[]> getKeyArray() {
+        return Optional.of(getKey().toByteArray().clone());
+    }
```

Comment:
* this should not be `public`
* there's no need to complicate the API with `Optional`
* there's no need to `clone()` here; `X509Key.getKey()` returns a copy
* suggest a better name would be `getKeyAsByteArray`

```java
    byte[] getKeyAsByteArray() {
        return getKey().toByteArray();
    }
```

Reply:
I added the `Optional` to be consistent with `XDHPrivateKeyImpl`'s method `getScalar`, which returns the key in byte array form. It, also, allows me to throw an exception without adding an extra if statement. I can change it though if you feel like that is too much of a complication.

Same thing goes for setting it as `public`. The method for getting `U` as a `BigInteger` is `public` and the value can be used to get the byte array, so we are not exposing any further information. But again, I can set it to `protected`or default, if necessary.

Name will be changed and `clone()` will be removed.

The method will be as follows:
```
public Optional<byte[]> getKeyAsByteArray() {
        return Optional.of(getKey().toByteArray());
}
```

Is that OK, or do you want me to go with your original suggestion?

Reply:
Because the method represents internal API, it need not follow conventions of public API, so I still think all of my suggestions are appropriate.

Reply:
Changed as per your suggestion.